multinet/util.py

```diff
@@ -12,15 +12,17 @@
 TEST_DATA_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__), "../test/data"))
 
 
-def filter_unwanted_keys(rows: Sequence[Dict]) -> Sequence[Dict]:
-    """Remove any unwanted keys from the documents."""
+def filter_unwanted_keys(row: Dict) -> Dict:
+    """Remove any unwanted keys from a document."""
     unwanted = {"_rev", "_id"}
-    new_rows = []
+    return {k: v for k, v in row.items() if k not in unwanted}
 
-    for row in rows:
-        new_rows.append({k: v for k, v in row.items() if k not in unwanted})
 
-    return new_rows
+def generate_filtered_docs(rows: Sequence[Dict]) -> Generator[Dict, None, None]:
+    """Filter unwanted keys from all documents with a generator."""
+
+    for row in rows:
+        yield filter_unwanted_keys(row)
```

Comment:
I think you can actually just write this as `(filter_unwanted_keys(row) for row in rows)` (note the parentheses in place of square brackets), in which case this function wouldn't be needed. But this is just an observation; no need to make that change here.